scripts/worldcover/run.py

```diff
@@ -0,0 +1,251 @@
+#!/usr/bin/env python3
+
+# import sys
+# sys.path.append("/home/tam/Documents/repos/model")
+
+import os
+import tempfile
+from math import floor
+
+import boto3
+import einops
+import geopandas as gpd
+import numpy
+import pyarrow as pa
+import rasterio
+import torch
+from rasterio.windows import Window
+from shapely import box
+from torchvision.transforms import v2
+
+from src.datamodule import ClayDataset
+from src.model_clay import CLAYModule
+
+YEAR = int(os.environ.get("YEAR", 2020))
+DATE = f"{YEAR}-06-01"
+TILE_SIZE = 12000
+CHIP_SIZE = 512
+E_W_INDEX_START = 67
+E_W_INDEX_END = 125
+N_S_INDEX_START = 24
+N_S_INDEX_END = 49
+YORIGIN = 50.0
+XORIGIN = -125.0
+PXSIZE = 8.333333333333333e-05
+
+RASTER_X_SIZE = (E_W_INDEX_END - E_W_INDEX_START) * TILE_SIZE
+RASTER_Y_SIZE = (N_S_INDEX_END - N_S_INDEX_START) * TILE_SIZE
+NODATA = 0
+CKPT_PATH = "s3://clay-model-ckpt/v0/mae_epoch-24_val-loss-0.46.ckpt"
+# CKPT_PATH = "https://huggingface.co/made-with-clay/Clay/resolve/main/Clay_v0.1_epoch-24_val-loss-0.46.ckpt"
+VERSION = "002"
+BUCKET = "clay-worldcover-embeddings"
+URL = "https://esa-worldcover-s2.s3.amazonaws.com/rgbnir/{year}/N{yidx}/ESA_WorldCover_10m_{year}_v{version}_N{yidx}W{xidx}_S2RGBNIR.tif"
+WC_VERSION_LOOKUP = {
+    2020: 100,
+    2021: 200,
+}
+
+
+MEAN = [
+    1369.03,  # red
+    1597.68,  # green
+    1741.10,  # blue
+    2858.43,  # nir
+]
+STD = [
+    2026.96,  # red
+    2011.88,  # green
+    2146.35,  # blue
+    2016.38,  # nir
+]
+
+grid = gpd.read_file(
+    # "/home/tam/Desktop/usa/esa_worldcover_grid_usa.fgb"
+    "https://clay-mgrs-samples.s3.amazonaws.com/esa_worldcover_grid_usa.fgb"
+)
+
+
+def tiles_and_windows(input: Window):
+    print("Input", input)
+    x_tile_index = E_W_INDEX_END - floor(input.col_off / TILE_SIZE)
+    x_local_off = input.col_off % TILE_SIZE
+    x_size = min(CHIP_SIZE, TILE_SIZE - x_local_off)
+    x_another = x_size < CHIP_SIZE
+
+    y_tile_index = N_S_INDEX_END - floor(input.row_off / TILE_SIZE)
+    y_local_off = input.row_off % TILE_SIZE
+    y_size = min(CHIP_SIZE, TILE_SIZE - y_local_off)
+    y_another = y_size < CHIP_SIZE
+
+    tile_id = f"N{y_tile_index}W{str(x_tile_index).zfill(3)}"
```

Comment:
Yes the path creation should probably be generalized. It was hardcoded here because the source file is for the US which is in the NW quadrant. 

Generalizing will take some tinkering to make sure that this still works even if you are looking at an area that crosses these regions.